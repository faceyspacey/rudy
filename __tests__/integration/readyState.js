import createTest from '../../__helpers__/createTest'

import {
  transformAction,
  call,
  enter
} from '../../src/middleware'

// createTest('ready === true when route has no thunks', {
//   SECOND: {
//     path: '/second',
//     onEnter: async (req) => {
//       expect(req.getLocation().ready).toEqual(true)
//     },
//     onComplete: async (req) => {
//       expect(req.getLocation().ready).toEqual(true)
//     }
//   }
// })

createTest('ready === true when route thunk response is cached', {
  SECOND: {
    path: '/second',
    beforeEnter: async (req) => {
      req.cache.cacheAction('thunk', req.action)
    },
    onEnter: async (req) => {
      expect(req.getLocation().ready).toEqual(true)
    },
    thunk: () => {
      return { foo: 'SUCCESS' }
    },
    onComplete: async (req) => {
      expect(req.getLocation().ready).toEqual(true)
    }
  }
})

// createTest('ready === false when route has non-cached thunk', {
//   SECOND: {
//     path: '/second',
//     beforeEnter: async (req) => {
//       console.log(req.getLocation().ready, req.route._isAsync)
//       // expect(req.getLocation().ready).toEqual(false)
//     },
//     thunk: async (req) => {
//       console.log(req.getLocation().ready)
//       expect(req.getLocation().ready).toEqual(false)
//       return { foo: 'SUCCESS' }
//     },
//     onComplete: async (req) => {
//       expect(req.getLocation().ready).toEqual(true)
//     }
//   }
// })
