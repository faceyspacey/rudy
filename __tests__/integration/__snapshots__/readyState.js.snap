// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ready === true when route thunk response is cached: 1 - response 1`] = `
Object {
  "basename": "",
  "cached": true,
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/second",
          "scene": "",
          "search": "",
          "url": "/second",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "SECOND",
      },
    ],
    "from": null,
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "scene": "",
    "search": "",
    "status": 200,
    "url": "/second",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "SECOND",
}
`;

exports[`ready === true when route thunk response is cached: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`ready === true when route thunk response is cached: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`ready === true when route thunk response is cached: 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`ready === true when route thunk response is cached: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`ready === true when route thunk response is cached: 1 - routes - SECOND - onEnter 1`] = `1`;

exports[`ready === true when route thunk response is cached: 1 - routes - SECOND - thunk 1`] = `0`;

exports[`ready === true when route thunk response is cached: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/second",
          "scene": "",
          "search": "",
          "url": "/second",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "SECOND",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "SECOND",
}
`;

exports[`ready === true when route thunk response is cached: 1 - title 1`] = `"SECOND"`;
